### PR TITLE
S143: Butterfly space

### PR DESCRIPTION
**Pull request by prabau**
S143 (Butterfly space) provides an example of cosmic space that is not an $\aleph_0$-space.

https://topology.pi-base.org/spaces?q=Cosmic%2B%7E%24%5Caleph_0%24-space

### CODE REVIEW

**Comment by ccaruvana** [comment]
Not sure, but would it be helpful to modify the justification of P182 to mirror the language used for the justification of P17? I agree it is clear what is meant by "Euclidean" in both cases, but the linking in P17 feels a bit cleaner.

**Comment by prabau** [comment]
> Not sure, but would it be helpful to modify the justification of P182 to mirror the language used for the justification of P17? I agree it is clear what is meant by "Euclidean" in both cases, but the linking in P17 feels a bit cleaner.

Makes sense. Let me make the change.

**Comment by prabau** [comment]
Hmm, the build is failing because it's missing a change from #762.  We can merge the `main` branch into this one to fix it.  Do you know how to do this?

**Comment by ccaruvana** [comment]
Thanks for the patience here as I navigate things. As I've been able, I have checked on each of the asserted properties here and I had one broad question. I know @StevenClontz has mentioned before to keep the doing of the math elsewhere, and I am not sure whether we would think something like the assertion for P23, though short, should be "aired out." Depending on the thoughts of others, if we deem it appropriate to take to stack exchange, I would be happy to post a generic question like "what other basic properties of the butterfly space are satisfied" and accept the relevant answer from another. However, it's also terribly unclear to me how to make that question somewhat beneficial to the stack exchange community.

I think what I'll do is just make a final few checks, approve this valuable addition, and then see if we should make modifications afterwards. So, again, thanks for the patience here.

**Comment by prabau** [comment]
My understanding was that it was not always the case but nowadays if a justification is short and simple enough it can be presented directly in pi-base.  Anything more complicated or subject to doubt or discussion should have a reference to the literature or a book or a post on mathse/mo.

Here I wrote two or three sentences for P23 that seemed clear enough.  But would appreciate if @StevenClontz can give his opinion.  We'll go from there.

**Comment by ccaruvana** [comment]
Also in the process here of looking things over, is the only topological difference between Michael's definition and Engelking's definition the x-axis in the sense that removing it from Michael's version doesn't disconnect the space but it does disconnect Engelking's version? The argument provided here for (local) arc connectedness is easily seen to have an adaptation for Engelking's definition, I guess just by modifying exponents to make sure whatever curve you have stays "below" the tangent circles.

**Comment by prabau** [comment]
> Also in the process here of looking things over, is the only topological difference between Michael's definition and Engelking's definition the x-axis in the sense that removing it from Michael's version doesn't disconnect the space but it does disconnect Engelking's version?

Yeah, the Engelking version considers all point of the plane, both above and below the x-axis.  So the "butterfly neighborhoods" in this case look like real bowties (symmetric wrt the x-axis), whereas the Michael version is only the upper half of it essentially.  Apart from that, most properties look pretty much the same.

> 
> The argument provided here for (local) arc connectedness is easily seen to have an adaptation for Engelking's definition, I guess just by modifying exponents to make sure whatever curve you have stays "below" the tangent circles. So I think all other properties here hold for Engelking's definition.

For the Engelking version one could use an arc with a high degree of tangency to the x-axis, like the graph of $e^{-x^2}$.

Note: Steen & Seebach also have the bowtie space on p. 175.  Their picture looks closer to the Engelking version maybe. I did not look at the original reference from MacAuley, as the Michael version seemed simpler to work with.

**Comment by prabau** [comment]
I have slightly expanded the explanation for P23 to make it clearer.

File: spaces/S000143/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+aliases:
+- Bow-tie space
+refs:
+  - zb: "0148.16701"
+    name: $\aleph_0$-spaces (E. Michael)
+  - zb: "0684.54001"
+    name: General Topology (Engelking, 1989)
+---
+
+The space $X$ is the closed upper half-plane topologized as follows.  Points above the $x$-axis have their usual Euclidean neighborhoods.  A local base for a point $p$ on the $x$-axis is formed by the "butterfly neighborhoods" $N_\varepsilon(p)$ for $\varepsilon>0$, where $N_\varepsilon(p)$ consists of $p$ together with all points $q$ at distance less than $\varepsilon$ from $p$ and lying underneath the union of the two rays in $X$ that emanate from $p$ and have slopes $\varepsilon$ and $-\varepsilon$.
```
Comment (StevenClontz) [inline spaces/S000143/README.md:13]:
```suggestion
The space $X=\mathbb R\times [0,\infty)$ is the closed upper half-plane topologized as follows. Points above the $x$-axis have their usual Euclidean neighborhoods. A local base for a point $p=(p_x,0)$ on the $x$-axis is formed by the "butterfly neighborhoods" $N_\varepsilon(p)$ for $\varepsilon>0$, where $$N_\varepsilon(p)=X\cap B_\varepsilon(p)\setminus\{(x,y):y\geq \varepsilon|x-p_x|\}$$ consists of $p$ together with all points $q$ at distance less than $\varepsilon$ from $p$ and lying underneath the union of the two rays in $X$ that emanate from $p$ and have slopes $\varepsilon$ and $-\varepsilon$.
```

image

(assuming I understood the definition)

**Comment by prabau** [comment]
@StevenClontz I thought pi-base did not understand double dollars `$$` for equations.  Has this been fixed? At least it accepts it in https://topology.pi-base.org/dev/preview.

File: spaces/S000143/README.md (hunk 1 above)
Comment (prabau) [reply spaces/S000143/README.md:13]: The suggested formula does not quite work, as it omits the point $p$.  But I'll fix this.

**Comment by prabau** [comment]
Tried the double dollars in the latest commit and it now works!  Nice.

**Comment by ccaruvana** [comment]
Great! Thanks @StevenClontz for the additional nod. I have a few class related things to finish up, but I should be finishing up my review of this PR today. Thanks again for everyone's patience.

**Review by ccaruvana — APPROVED** [review]

**Comment by StevenClontz** [comment]
> @StevenClontz I thought pi-base did not understand double dollars `$$` for equations. Has this been fixed? At least it accepts it in https://topology.pi-base.org/dev/preview.

I think this might have slipped in with the recent (I guess it was maybe a year ago now?) refactor of the pi-Base app.

**Pull request merged**